### ISSUE TITLE
ERROR: No resource or method named `git_client' for `Chef::Recipe "default"'

### ISSUE THREAD

**Issue by gregelin** [fisma-ready/ubuntu-lts#37]
On a first time attempt to run instructions, receiving error on amazon-ebs that git cookbook is missing.

Excerpted debug output:

```
Gregs-MBP:ubuntu-lts greg$ packer build -debug \
>   -var "aws_access_key_id=$AWS_ACCESS_KEY_ID" \
>   -var "aws_secret_key=$AWS_SECRET_ACCESS_KEY" \
>   -var "grub_passwd=$GRUB_PWD" \
>   packer.json
Debug mode enabled. Builds will not be parallelized.
amazon-ebs output will be in this color.


amazon-ebs:
    amazon-ebs: Creating directory: /tmp/packer-chef-solo
    amazon-ebs: Creating directory: /tmp/packer-chef-solo/cookbooks-0
    amazon-ebs: Creating configuration file 'solo.rb'
    amazon-ebs: Creating JSON attribute file
    amazon-ebs: Executing Chef: sudo chef-solo --no-color -c /tmp/packer-chef-solo/solo.rb -j /tmp/packer-chef-solo/node.json
    amazon-ebs: Starting Chef Client, version 12.3.0
    amazon-ebs: Compiling Cookbooks...
    amazon-ebs: [2015-06-02T10:36:43+00:00] WARN: MissingCookbookDependency:
    amazon-ebs: Recipe `git` is not in the run_list, and cookbook 'git'
    amazon-ebs: is not a dependency of any cookbook in the run_list.  To load this recipe,
    amazon-ebs: first add a dependency on cookbook 'git' in the cookbook you're
    amazon-ebs: including it from in that cookbook's metadata.
    amazon-ebs:
    amazon-ebs:
    amazon-ebs: ================================================================================
    amazon-ebs: Recipe Compile Error in /tmp/packer-chef-solo/cookbooks-0/packer/recipes/default.rb
    amazon-ebs: ================================================================================
    amazon-ebs:
    amazon-ebs: NoMethodError
    amazon-ebs: -------------
    amazon-ebs: No resource or method named `git_client' for `Chef::Recipe "default"'
    amazon-ebs:
    amazon-ebs: Cookbook Trace:
    amazon-ebs: ---------------
    amazon-ebs: /tmp/packer-chef-solo/cookbooks-0/git/recipes/default.rb:19:in `from_file'
    amazon-ebs: /tmp/packer-chef-solo/cookbooks-0/packer/recipes/default.rb:6:in `from_file'
    amazon-ebs:
    amazon-ebs: Relevant File Content:
    amazon-ebs: ----------------------
    amazon-ebs: /tmp/packer-chef-solo/cookbooks-0/git/recipes/default.rb:
    amazon-ebs:
    amazon-ebs: 12:  #
    amazon-ebs: 13:  # Unless required by applicable law or agreed to in writing, software
    amazon-ebs: 14:  # distributed under the License is distributed on an "AS IS" BASIS,
    amazon-ebs: 15:  # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    amazon-ebs: 16:  # See the License for the specific language governing permissions and
    amazon-ebs: 17:  # limitations under the License.
    amazon-ebs: 18:
    amazon-ebs: 19>> git_client 'default' do
    amazon-ebs: 20:    action :install
    amazon-ebs: 21:  end
    amazon-ebs: 22:
    amazon-ebs:
    amazon-ebs:
    amazon-ebs: Running handlers:
    amazon-ebs: [2015-06-02T10:36:43+00:00] ERROR: Running exception handlers
    amazon-ebs: Running handlers complete
    amazon-ebs: [2015-06-02T10:36:43+00:00] ERROR: Exception handlers complete
    amazon-ebs: [2015-06-02T10:36:43+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
    amazon-ebs: Chef Client failed. 0 resources updated in 1.415970956 seconds
    amazon-ebs: [2015-06-02T10:36:44+00:00] ERROR: No resource or method named `git_client' for `Chef::Recipe "default"'
    amazon-ebs: [2015-06-02T10:36:44+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
    amazon-ebs:
    amazon-ebs:
==> amazon-ebs: Pausing before cleanup of step 'StepConnectSSH'. Press enter to continue. 
==> amazon-ebs: Pausing before cleanup of step 'StepRunSourceInstance'. Press enter to continue. 
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Pausing before cleanup of step 'StepSecurityGroup'. Press enter to continue. 
==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Pausing before cleanup of step 'StepKeyPair'. Press enter to continue. 
==> amazon-ebs: Deleting temporary keypair...
==> amazon-ebs: Pausing before cleanup of step 'StepSourceAMIInfo'. Press enter to continue. 
Build 'amazon-ebs' errored: Error executing Chef: Non-zero exit status: 1

```

**Comment by roacobb** [comment]
@gregelin what packer version do you have? Also what chef version are you running?

**Comment by gregelin** [comment]
@roacobb  Installed both Chef and Packer today:

```
$ chef --version
Chef Development Kit Version: 0.6.0
chef-client version: 12.3.0
berks version: 3.2.4
kitchen version: 1.4.0

$packer version
Packer v0.7.5
```

**Comment by ozzyjohnson** [comment]
This looks to be an incompatibility with 4.2 versions of the git cookbook and the newer chef-client.

Pinning git to 4.1.0 in `Berksfile` is an immediate workaround.

`cookbook 'git', '4.1.0'`

**Comment by gregelin** [comment]
@ozzyjohnson @roacobb 
Thanks for help. That seems to work.

- How did you figure that out?
- Do you want me to make a pull request?

**Comment by gregelin** [comment]
@ozzyjohnson @roacobb Just noticed this also seems related to a pull request and has a different solution metadata.rb. https://github.com/fisma-ready/ubuntu-lts/pull/36

**Comment by gregelin** [comment]
Fixed by Pull Request for #36
